WhoIsThat/WhoIsThat/HomePage.xaml.cs

```diff
@@ -9,12 +12,52 @@
 
 namespace WhoIsThat
 {
+
 	[XamlCompilation(XamlCompilationOptions.Compile)]
 	public partial class HomePage : ContentPage
 	{
 		public HomePage ()
 		{
 			InitializeComponent ();
 		}
+
+        private async void takePicture(object sender, EventArgs e)
+        {
+            //Requesting permissions to user camera and storage, too lazy to implement checks if user already have those, later 'todo'
+            await CrossPermissions.Current.RequestPermissionsAsync(Permission.Camera);
+            await CrossPermissions.Current.RequestPermissionsAsync(Permission.Storage);
+
+            //Initializing media hardware
+            await CrossMedia.Current.Initialize();
+
+            //Checking if camera is available, if not -> return
+            if (!CrossMedia.Current.IsCameraAvailable || !CrossMedia.Current.IsTakePhotoSupported)
+            {
+                await DisplayAlert("No camera", "No camera available.", "OK");
```

Comment:
This could be achieved through exception or an event, in order not to couple camera and views (alerts)